custom_components/lennoxs30/config_flow.py

```diff
@@ -0,0 +1,110 @@
+import ipaddress
+import re
+
+import voluptuous as vol
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_EMAIL,
+    CONF_HOSTS,
+    CONF_IP_ADDRESS,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.helpers import config_validation as cv
+import logging
+
+CONF_FAST_POLL_INTERVAL = "fast_scan_interval"
+CONF_ALLERGEN_DEFENDER_SWITCH = "allergen_defender_switch"
+CONF_APP_ID = "app_id"
+CONF_INIT_WAIT_TIME = "init_wait_time"
+CONF_CREATE_SENSORS = "create_sensors"
+DEFAULT_POLL_INTERVAL: int = 10
+DEFAULT_FAST_POLL_INTERVAL: float = 0.75
+MAX_ERRORS = 5
+RETRY_INTERVAL_SECONDS = 60
+DOMAIN = "lennoxs30"
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_NAME, default = "S30"): cv.string,
+                vol.Required(CONF_EMAIL): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                vol.Required(CONF_HOSTS, default="Cloud"): str,
+                vol.Optional(CONF_SCAN_INTERVAL): cv.positive_int,
+                vol.Optional(
+                    CONF_FAST_POLL_INTERVAL, default=DEFAULT_FAST_POLL_INTERVAL
+                ): cv.positive_float,
+                vol.Optional(CONF_ALLERGEN_DEFENDER_SWITCH, default=False): cv.boolean,
+                vol.Optional(CONF_APP_ID): cv.string,
+                vol.Optional(CONF_INIT_WAIT_TIME, default=30): cv.positive_int,
+                vol.Optional(CONF_CREATE_SENSORS, default=False): cv.boolean,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+def host_valid(host):
+    """Return True if hostname or IP address is valid."""
```

Comment:
A detail, which we can fix later.  

I’d like this to support a host:port syntax.  Two use cases, first I’m putting all my IOT devices on a subnet behind a router / firewall and will use port forwarding to interface to.  So instead of 10.0.0.1 port 443 (default https port), it may be 10.0.0.1:65433 which the router/firewall can forward to the S30. 

Second use case, I’ve decided to start on an S30 simulator, so we’d want to be able to run multiple on the same host with different ports.